server/api.js

```diff
@@ -134,4 +134,97 @@ router.get("/tenders", async (req, res) => {
 	}
 });
 
+router.get("/bid", async (req, res) => {
+	const tenderID = parseInt(req.query.tender_id);
+	let page = parseInt(req.query.page) || 1;
+	const itemsPerPage = 10;
+
+	const totalBidsPerTender = await db.query(
+		"SELECT COUNT(tender_id) FROM bid WHERE tender_id = $1",
+		[tenderID]
+	);
+	const totalPages = Math.ceil(totalBidsPerTender.rows[0].count / itemsPerPage);
+	const offset = (page - 1) * itemsPerPage;
+
+	const totalBidsResults = await db.query(
+		"SELECT * FROM bid WHERE tender_id = $1 LIMIT $2 OFFSET $3",
+		[tenderID, itemsPerPage, offset]
+	);
+
+	totalBidsResults
+		? res.send({
+				results: totalBidsResults.rows,
+				pagination: {
+					itemsPerPage: 10,
+					currentPage: page,
+					totalPages: totalPages,
+				},
+		  })
+		: res.status(500).send({ code: "SERVER_ERROR" });
+});
+
+router.post("/bid/:bidId/status", async (req, res) => {
+	const bidId = parseInt(req.params.bidId, 10);
+	const status = req.body.status;
+	const validStatuses = ["Awarded", "Rejected", "Withdraw", "In review"];
+
+	if (!validStatuses.includes(status)) {
+		return res.status(400).send({ code: "INVALID_STATUS" });
+	}
+
+	try {
+		await db.query("BEGIN");
```

Comment:
The documentation recommends using `pool.connect()` . Let's do the same to be safe. You can look at hos Aisha did it here: https://github.com/LauraSantiag0/Love-Me-Tender/pull/44/files